components/PralineBox/index.js

```diff
@@ -0,0 +1,170 @@
+import { useRouter } from "next/router";
+import React, { useState } from "react";
+import Image from "next/image";
+import InputField from "../PralineForm/InputField";
+import { Print, StyledBox, StyledWrapper } from "./box.styled";
+import Modal from "../Modal/Modal";
+import PralineList from "../PralineList/PralineList";
+
+export default function MainPage() {
+  const router = useRouter();
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [slotId, setSlotId] = useState(null);
+  const [boxsize, setBoxsize] = useState(9);
+  const [pralineList, setPralineList] = useState([]);
+  const [pralineBoxName, setPralineBoxName] = useState("");
+
+  function cancel() {
+    setPralineList([]);
+    setPralineBoxName("");
+  }
+
+  const ingredientSum = pralineList
+    .map((praline) => praline.ingredients)
+    .flat()
+    .reduce(
+      (sum, current) => ({
+        ...sum,
+        [current.ingredient]:
+          (sum[current.ingredient] || 0) + parseInt(current.amount, 10),
+      }),
+      {}
+    );
+
+  const ingredientList = Object.entries(ingredientSum)
+    .map(([name, amount]) => ({
+      name,
+      amount,
+    }))
+    .sort((a, b) => (a.amount > b.amount ? -1 : 1));
+
+  const allergySum = pralineList
+    .map((praline) => praline.allergyTraces)
+    .flat()
+    .reduce(
+      (sum, current) => ({
+        ...sum,
+        [current.ingredient]:
+          (sum[current.ingredient] || 0) + parseInt(current.amount, 10),
+      }),
+      {}
+    );
+
+  const allergyList = Object.entries(allergySum)
+    .map(([name, amount]) => ({
+      name,
+      amount,
+    }))
+    .sort((a, b) => (a.amount > b.amount ? -1 : 1));
+
+  function handleAddPraline(newpraline) {
+    const newArray = pralineList;
+    newArray[slotId] = newpraline;
+    setPralineList(newArray);
+  }
+
+  const weightSum = pralineList.reduce(
+    (sum, current) => sum + parseFloat(current.weight, 10),
+    0
+  );
+
+  const getBoxes = () => {
+    const boxes = [];
+
+    for (let i = 0; i < boxsize; i += 1) {
+      const key = i;
+      boxes.push(
+        <StyledBox
+          key={key}
```

Comment:
The `index` is not enough for the key prop. You need a stable and unique identifier.